api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -532,7 +533,14 @@ public void setVisit(Visit visit) {
 		
 		return encounterProviders.stream()
 				.filter(ep -> includeVoided || !ep.getVoided())
-				.collect(Collectors.groupingBy(EncounterProvider::getEncounterRole, Collectors.mapping(EncounterProvider::getProvider, Collectors.toSet())));
+				.collect(Collectors.toMap((ep) -> (ep).getEncounterRole(),
```

Comment:
@djazayeri can you please join our discussion. by making use of streams here we ran into a NPE (due to a java bug) which @pushpa446 would fix with this change. see explanation starting from my 2 post. but this also lead to @lluismf fundamental question of why null is allowed in in the first place.